### PR TITLE
Cache headers for manifests.

### PR DESCRIPTION
**Pull request by RichardScothern**
Add Etag header for manifests.  Return 304 (Not Modified) if retrieved with If-None-Match header

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

### CODE REVIEW

File: registry/handlers/images.go [hunk 1]
```diff
@@ -160,6 +178,7 @@ func (imh *imageManifestHandler) PutImageManifest(w http.ResponseWriter, r *http
 
 	w.Header().Set("Location", location)
 	w.Header().Set("Docker-Content-Digest", imh.Digest.String())
+	w.Header().Set("Etag", imh.Digest.String())
```
Comment (stevvooe) [inline registry/handlers/images.go]: We don't return the manifest here, so the Etag is not necessary.

File: registry/handlers/images.go [hunk 2]
```diff
@@ -60,6 +60,10 @@ func (imh *imageManifestHandler) GetImageManifest(w http.ResponseWriter, r *http
 	if imh.Tag != "" {
 		sm, err = manifests.GetByTag(imh.Tag)
 	} else {
+		if etagMatch(r, imh.Digest.String()) {
```
Comment (stevvooe) [inline registry/handlers/images.go]: This is slightly tautological... but correct.

Comment (RichardScothern) [inline registry/handlers/images.go]: Me too.

**Comment by stevvooe** [comment]
LGTM

**Pull request merged**